rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp

```diff
@@ -201,13 +286,15 @@ __set_parameters_atomically_common(
   if (on_set_parameters_callback) {
     result = on_set_parameters_callback(parameters);
   }
+  result.successful =
+    result.successful && __check_parameters(parameter_infos, parameters);
 
   // If accepted, actually set the values.
   if (result.successful) {
     for (size_t i = 0; i < parameters.size(); ++i) {
       const std::string & name = parameters[i].get_name();
-      parameter_infos[name].descriptor.name = parameters[i].get_name();
       parameter_infos[name].descriptor.type = parameters[i].get_type();
+      parameter_infos[name].descriptor.name = parameters[i].get_name();
```

Comment:
This is really a detail, but given that the name shouldn't change after the descriptor is set the first time, maybe it's not necessary to set the name of the descriptor every time a parameter is set.

Reply:
Yeah, I agree. The original code was doing that.

Reply:
Maybe this diff should be eliminated? Not sure what putting the assignment of the name below the assignment of the type does, but I think nothing.

Reply:
Sorry, didn't realize that I moved it. At some point I deleted it, and then I added it again in the wrong line.

Reply:
Addressed in https://github.com/ros2/rclcpp/pull/735/commits/2755108e124d6ceb7617bd90f32784c2a5d3af66.